workflows/genomeassembler.nf

```diff
@@ -68,16 +113,188 @@ workflow GENOMEASSEMBLER {
 
     ch_versions = Channel.empty()
 
+    // Check stage
+    def workflow_steps = params.step.tokenize(",")
+    // Define constants - TODO: Move to JSON validation
+    def workflow_permitted_stages = ['data_qc','preprocess','assemble','validate','curate']
+    if ( ! workflow_steps.every { it in workflow_permitted_stages } ) {
+        error "Unrecognised workflow step in $params.step ( $workflow_permitted_stages )"
+    }
+
     //
     // SUBWORKFLOW: Read in samplesheet, validate and stage input files
-    //
-    // INPUT_CHECK (
-    //     ch_input
-    // )
     PREPARE_INPUT (
         ch_input
     )
-    // ch_versions = ch_versions.mix(PREPARE_INPUT.out.versions)
+    ch_versions = ch_versions.mix ( PREPARE_INPUT.out.versions )
+
+    // - TODO: Fetch Genome properties from GOAT
+
+    // BUILD KMER DATABASES
+    // builds k-mer databases as a separate step to allow reuse with resume
+    if ( params.kmer_counter == 'meryl' && ['data_qc','validate'].any { it in workflow_steps } ) {
+        BUILD_HIFI_MERYL_DATABASE ( PREPARE_INPUT.out.hifi )
+        BUILD_HIC_MERYL_DATABASE ( PREPARE_INPUT.out.hic)
+        BUILD_ILLUMINA_MERYL_DATABASE ( PREPARE_INPUT.out.illumina )
+        if( params.enable_ont_kmer_analyses ) {
```

Comment:
Yes, because ONT data is generally too noisy for k-mer analyses to apply. It's only in the current unreleased chemistry that it might be accurate enough to apply k-mer analyses. I guess one could apply read correction first and then run it through here, but I'll leave that for someone else to code if and when.